beacon_node/lighthouse_network/src/rpc/codec.rs

```diff
@@ -596,10 +597,12 @@ fn handle_rpc_request<E: EthSpec>(
         ))),
         SupportedProtocol::DataColumnsByRootV1 => Ok(Some(RequestType::DataColumnsByRoot(
             DataColumnsByRootRequest {
-                data_column_ids: RuntimeVariableList::from_ssz_bytes(
-                    decoded_buffer,
-                    spec.max_request_data_column_sidecars as usize,
-                )?,
+                data_column_ids:
+                    <RuntimeVariableList<DataColumnsByRootIdentifier>>::from_ssz_bytes_with_nested(
+                        decoded_buffer,
+                        spec.max_request_blocks(current_fork),
```

Comment:
This is inconsistent with
https://github.com/sigp/lighthouse/blob/d54bffcb787d957b257ea5abd257674118674c95/beacon_node/lighthouse_network/src/rpc/methods.rs#L489

The spec says Deneb but I think current_fork would be compliant too and forward compatible. We do the same  for `max_request_blob_sidecars` above too. I'll update the other spot to use `current_fork`.

Reply:
Updated in https://github.com/sigp/lighthouse/pull/7399/commits/7b0e862c93fb3c2ae307f71ab1f1a976436e50ff